src/connector/src/source/filesystem/opendal_source/mod.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod gcs_source;
+
+pub use gcs_source::*;
+pub mod s3_source;
+pub use s3_source::*;
+use serde::Deserialize;
+pub mod opendal_enumerator;
+pub mod opendal_reader;
+
+use self::opendal_enumerator::OpendalEnumerator;
+use self::opendal_reader::OpendalReader;
+use super::{OpendalFsSplit, S3Properties};
+use crate::source::SourceProperties;
+
+pub const GCS_CONNECTOR: &str = "gcs";
+// The new s3_v2 will use opendal.
+pub const OPENDAL_S3_CONNECTOR: &str = "s3_v2";
```

Comment:
This is a little confusing to me (without the background). IIUC, s3 v2 refers to the new algorithm, while opendal refers to an orthogonal aspect. 

Could you tell me the reasons to use opendal for s3 v2, or points me to related discussions?

BTW, I think using `S3_V2` as the variable name might be clearer than `OPENDAL_S3`.